### PR TITLE
net: wifi: Add API docs

### PR DESCRIPTION
**Pull request by krish2718**
Add a basic API docs, this can be extended to add detailed sections and details in future PRs.

Based on #59249 as that has few structure changes.

### CODE REVIEW

File: include/zephyr/net/wifi_mgmt.h [hunk 1]
```diff
@@ -144,111 +174,167 @@ enum net_event_wifi_cmd {
...
+	/** Frequency band */
 	uint8_t band;
+	/** Channel */
 	uint8_t channel;
+	/** Security type */
 	enum wifi_security_type security;
+	/** MFP options */
 	enum wifi_mfp_options mfp;
-	int timeout; /* SYS_FOREVER_MS for no timeout */
+	/** Connect timeout in milliseconds, SYS_FOREVER_MS for no timeout */
```
Comment (krish2718) [inline include/zephyr/net/wifi_mgmt.h:235]: This is actually seconds, not milliseconds.

File: doc/connectivity/networking/api/wifi.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Overview
+========
+
+The Wi-Fi management API is used to manage Wi-Fi networks. It supports below modes:
+
+* IEEE802.11 Station (STA)
+* IEEE802.11 Access Point (AP)
+
+The Wi-Fi management API is implemented in the `wifi_mgmt` module as a networking L2 stack.
```
Comment (rlubos) [inline doc/connectivity/networking/api/wifi.rst:14]: > implemented in the `wifi_mgmt` module as a networking L2 stack

This is a bit unclear to me, can you elaborate on what did you mean here exactly? What does it mean that "wifi_mgmt" is implemented as a "networking L2 stack"?

**Review by rlubos — COMMENTED** [review]
Overall looks good, just one question.

File: doc/connectivity/networking/api/wifi.rst (hunk 2 above)
Comment (krish2718) [reply doc/connectivity/networking/api/wifi.rst:14]: By `stack` I was referring to an entity part of kernel and sitting on top of L2 which for Wi-Fi could be Etherne/Offloaded

File: doc/connectivity/networking/api/wifi.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+################
+
+Overview
+========
+
+The Wi-Fi management API is used to manage Wi-Fi networks. It supports below modes:
+
+* IEEE802.11 Station (STA)
+* IEEE802.11 Access Point (AP)
+
```
Comment (rlubos) [inline doc/connectivity/networking/api/wifi.rst:13]: So maybe just mention that `wifi_mgmt` is just a part of the L2:
```suggestion
The Wi-Fi management API is implemented in the `wifi_mgmt` module as a part of the networking L2 stack.
```

File: doc/connectivity/networking/api/wifi.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+The Wi-Fi management API is implemented in the `wifi_mgmt` module as a networking L2 stack.
+Currently, two Wi-Fi drivers are supported:
```
Comment (rlubos) [inline doc/connectivity/networking/api/wifi.rst:15]:
```suggestion
Currently, two types of Wi-Fi drivers are supported:
```

**Review by rlubos — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: include/zephyr/net/wifi.h [hunk 5]
```diff
@@ -314,49 +370,70 @@ static const char * const wifi_twt_negotiation_type2str[] = {
...
+	/* TWT setup grouping (grouping of TWT flows) */
 	WIFI_TWT_SETUP_CMD_GROUPING,
+	/* TWT setup accept (parameters accepted by AP) */
 	WIFI_TWT_SETUP_CMD_ACCEPT,
+	/* TWT setup alternate (alternate parameters suggested by AP) */
 	WIFI_TWT_SETUP_CMD_ALTERNATE,
+	/* TWT setup dictate (parameters dictated by AP) */
 	WIFI_TWT_SETUP_CMD_DICTATE,
+	/* TWT setup reject (parameters rejected by AP) */
 	WIFI_TWT_SETUP_CMD_REJECT,
```
Comment (kartben) [inline include/zephyr/net/wifi.h:392]: Missing javadoc style comment

**Review by kartben — COMMENTED** [review]
I realize this has been merged already, but somehow I missed the chance to review in time. Any chance you could address in a follow up PR? @krish2718

**Comment by krish2718** [comment]
Raised: https://github.com/zephyrproject-rtos/zephyr/pull/60406

**Pull request merged**